pkg/operator/v1helpers/helpers.go

```diff
@@ -310,3 +313,30 @@ func MapToEnvVars(mapEnvVars map[string]string) []corev1.EnvVar {
 	sort.Slice(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
 	return envVars
 }
+
+// InjectObservedProxyIntoContainers returns a patched container slice with proxy env vars set.
+func InjectObservedProxyIntoContainers(containers []corev1.Container, observedConfig []byte, fields ...string) ([]corev1.Container, error) {
+	var config map[string]interface{}
+	if err := yaml.Unmarshal(observedConfig, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal the observedConfig: %w", err)
+	}
+
+	proxyConfig, found, err := unstructured.NestedStringMap(config, fields...)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get the proxy config from observedConfig: %w", err)
+	}
+
+	if !found {
+		return containers, nil
+	}
+
+	proxyEnvVars := MapToEnvVars(proxyConfig)
+	patchedContainers := []corev1.Container{}
+	for i := range containers {
```

Comment:
please use the annotation from https://github.com/openshift/cluster-version-operator/pull/224 to identify the containers to inject.

Reply:
done